darts/dataprocessing/transformers/midas.py

```diff
@@ -0,0 +1,328 @@
+
+        # make multiple low frequency columns out of the high frequency column(s)
+        midas_df = _create_midas_df(
+            series_df=series_df,
+            low_index_datetime=low_index_datetime,
+        )
+
+        # back to TimeSeries
+        midas_ts = TimeSeries.from_dataframe(
+            midas_df,
+            static_covariates=series.static_covariates,
+        )
+        if strip:
+            midas_ts = midas_ts.strip()
+
+        # components: comp0_0, comp1_0, comp0_1, comp1_1, ...
+        return midas_ts
+
+    @staticmethod
+    def ts_inverse_transform(
+        series: TimeSeries,
+        params: Mapping[str, Any],
+    ) -> TimeSeries:
+        """
+        Transforms series back to high frequency by retrieving the original high frequency and reshaping the values.
+        When converting to/from anchorable offset [1]_, the original time index is not garanteed to be restored.
+
+        Steps:
+            (1) Reshape the values to flatten the components introduced by the transform
+            (2) Eliminate the rows filled with NaNs, to facilitate time index adjustment
+            (3) Retrieve the original components name
+            (4) When applicable, shift the time index start back in time
+            (5) Generate a new time index with the high frequency
+
+        References
+        ----------
+        .. [1] https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#anchored-offsets
+        """
+        MIDAS._verify_series(series)
+        high_freq_name = params["fitted"]["high_freq_name"]
+        orig_ts_start_time = params["fitted"]["start"]
+        orig_ts_end_time = params["fitted"]["end"]
+
+        # retrieve the number of component introduced by midas
+        n_midas_components = int(series.components[-1].split("_")[-1]) + 1
+        series_n_components = series.n_components
+
+        # original ts was univariate
+        if n_midas_components == series_n_components:
+            n_orig_components = 1
+            series_values = series.values().flatten()
+        else:
+            n_orig_components = series_n_components // n_midas_components
+            series_values = series.values().reshape((-1, n_orig_components))
+
+        # retrieve original components name by removing the "_0" suffix
+        component_names = [
+            series.components[i][:-2] for i in range(0, n_orig_components)
+        ]
+
+        # remove the rows containing NaNs at the extremes, necessary to adjust the time index
+        first_finite_row = pd.core.missing.find_valid_index(series_values, how="first")
+        last_finite_row = pd.core.missing.find_valid_index(series_values, how="last")
+        series_values = series_values[first_finite_row : last_finite_row + 1]
+
+        start_time = series.start_time()
+        shift = 0
+        # adjust the start if was shifted due to the frequency change
+        if len(series.time_index) > 1:
```

Comment:
It's a bit tricky to understand this part.
does it only work if either the `series` starts at the same time as original series, or it ends one time step after the end of the original series?

Ideally this should work for any point in time